neo/Network/P2P/Peer.cs

```diff
@@ -179,25 +183,41 @@ private void OnTcpConnected(IPEndPoint remote, IPEndPoint local)
             ImmutableInterlocked.Update(ref ConnectingPeers, p => p.Remove(remote));
             if (MaxConnections != -1 && ConnectedPeers.Count >= MaxConnections && !TrustedIpAddresses.Contains(remote.Address))
             {
-                Sender.Tell(Tcp.Abort.Instance);
+                DisconnectWithAddresses(DisconnectReason.MaxConnectionReached, "The maximum number of connections reached!");
                 return;
             }
 
             ConnectedAddresses.TryGetValue(remote.Address, out int count);
             if (count >= MaxConnectionsPerAddress)
             {
-                Sender.Tell(Tcp.Abort.Instance);
+                DisconnectWithAddresses(DisconnectReason.MaxConnectionPerAddressReached, "The maximum number of per address connections reached!");
+                return;
             }
             else
             {
                 ConnectedAddresses[remote.Address] = count + 1;
                 IActorRef connection = Context.ActorOf(ProtocolProps(Sender, remote, local), $"connection_{Guid.NewGuid()}");
                 Context.Watch(connection);
                 Sender.Tell(new Tcp.Register(connection));
-                ConnectedPeers.TryAdd(connection, remote);
+                ConnectedPeers.TryAdd(connection, new List<IPEndPoint> { remote });
             }
         }
 
+        private void DisconnectWithAddresses(DisconnectReason reason, string message)
+        {
+            NetworkAddressWithTime[] networkAddresses = GetRandomConnectedPeers(AddrPayload.MaxCountToSend);
+            Disconnect(reason, message, networkAddresses.ToByteArray());
+        }
+
+        private void Disconnect(DisconnectReason reason, string message = "", byte[] data = null)
+        {
+            var payload = DisconnectPayload.Create(reason, message, data);
+            var disconnect = Message.Create(MessageCommand.Disconnect, payload);
+            var command = Tcp.Write.Create(ByteString.FromBytes(disconnect.ToArray()));
+            Sender.Tell(command);
+            Sender.Tell(Tcp.Close.Instance);
```

Comment:
Does AKKA guarantee the order?

Reply:
I'll correct it.

Reply:
Akka guarantee the message order. https://doc.akka.io/docs/akka/2.5/general/message-delivery-reliability.html

Note:
It is important to note that Akka’s guarantee applies to the order in which messages are enqueued into the recipient’s mailbox. If the mailbox implementation does not respect FIFO order (e.g. a PriorityMailbox), then the order of processing by the actor can deviate from the enqueueing order.

Reply:
fixed, plz have a look